### PR TITLE
Feature/task 97 enable and disable val

### LINKED ISSUE #97: Enable and Disable Validation Rules
There might be a need for enabling and disabling of validation rules.

At the moment we have all validations active for one concept. E.g. a check that IFEs are typed correctly and that IF are connected correctly to IFEs.

 1. In the future each rule should be implemented individually and have its own id
 2. Project specific settings should state which one of these rules is activated in the project, will be needed for continous system integration on a jenkins later.
 3.  Users should eb able to override the project settings with some user preferences, that one should be based on the eclipse preferences page dialog etc.

### PR DESCRIPTION
**Pull request by andriikovalov-dlr**
- Added a field suppressedValidators to Repository (collection of Strings validator class names)
- Moved getting applicable validators for a repository to a separate class
- Added a Repository snippet to add and remove suppressed validators

Closes #97, #300

### CODE REVIEW

File: de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/RepoValidatorsInstantiator.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	 * Reads all registered validator extensions and creates validator objects for applicable validators.
+	 */
+	private void createValidators() {
+		repoValidators = new ArrayList<>();
+		seiValidators = new ArrayList<>();
+		
+		Set<String> activeConceptIds = repository.getActiveConcepts().stream()
+				.map(c -> c.getFullQualifiedName()).collect(Collectors.toSet());
+		
+		Set <String> suppressedValidators = new HashSet<>(repository.getSuppressedValidators());
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/RepoValidatorsInstantiator.java]: Superflous whitespace after Set

File: de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/RepoValidatorsInstantiator.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	private Repository repository;
+	private List<IStructuralElementInstanceValidator> seiValidators;
+	private List<IRepositoryValidator> repoValidators;
+
+	public RepoValidatorsInstantiator(Repository repository) {
+		this.repository = repository;
+	}
+	
+	public List<IRepositoryValidator> getRepoValidators() {
+		if (repoValidators == null) {
```
[the remark this answers is not in the document]
Comment (andriikovalov-dlr) [reply de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/RepoValidatorsInstantiator.java]: Not really, I just didn't want to do heavy stuff in the constructor. We can change it.

File: de.dlr.sc.virsat.model/src-gen/de/dlr/sc/virsat/model/dvlm/categories/ATypeInstance.java [hunk 3]
```diff
@@ -70,7 +70,6 @@
 	 * Call this method to get a CategoryAssignmentContainer handed back for any TypeInstance. This method returns null in case it could not be found.
 	 * <!-- end-model-doc -->
 	 * @model kind="operation"
-	 *        annotation="http://www.eclipse.org/emf/2002/GenModel body='return de.dlr.sc.virsat.model.dvlm.categories.util.CategoryAssignmentHelper.getContainerFor(this);'"
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.model/src-gen/de/dlr/sc/virsat/model/dvlm/categories/ATypeInstance.java]: Hey @andriikovalov-dlr , why did all these annotations dissappear?

File: de.dlr.sc.virsat.model/model/dvlm.genmodel [hunk 4]
```diff
@@ -37,7 +38,8 @@
         <genFeatures property="None" children="true" createChild="true" ecoreFeature="ecore:EReference dvlm.ecore#//structural/StructuralElementInstance/relationInstances"/>
         <genFeatures property="None" children="true" createChild="true" ecoreFeature="ecore:EReference dvlm.ecore#//structural/StructuralElementInstance/children"/>
         <genFeatures property="None" notify="false" createChild="false" ecoreFeature="ecore:EReference dvlm.ecore#//structural/StructuralElementInstance/parent"/>
-        <genOperations ecoreOperation="dvlm.ecore#//structural/StructuralElementInstance/getDeepChildren"/>
+        <genOperations ecoreOperation="dvlm.ecore#//structural/StructuralElementInstance/getDeepChildren"
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.model/model/dvlm.genmodel]: What are these changes about? They seem to be connected to the missing annotations, aren't they.

File: de.dlr.sc.virsat.model/src-gen/de/dlr/sc/virsat/model/dvlm/categories/ATypeInstance.java (hunk 3 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.model/src-gen/de/dlr/sc/virsat/model/dvlm/categories/ATypeInstance.java]: Discussed together with @andriikovalov-dlr . This change could be a result of migrating to a newer eclipse and changes in EMD. Nevertheless, release notes of EMF do not give evidence. Besides that the annotations are now written to the genmodel rather the generated interfaces, nothing has changed.

Comment (andriikovalov-dlr) [reply de.dlr.sc.virsat.model/src-gen/de/dlr/sc/virsat/model/dvlm/categories/ATypeInstance.java]: It looks like the new EMF code generator doesn't put these annotations into javadoc (like here), but includes it into .genmodel

**Pull request merged**